### PR TITLE
[maintenance] Remove unnecessary base class in `onedal.cluster.KMeans`

### PR DESCRIPTION
**Pull request by icfaust**
## Description

The `_BaseKMeans` class is unnecessary and duplicates calls to `support_queue` unnecessarily.  It is simply deleted. Additional sklearn conformance class methods (i.e. like transform) are removed as they go unused. Note the `copy_x` keyword exists as a parameter in sklearn, but is neither set in sklearnex for the onedal estimator nor used at all in the onedal estimator.

No loss in functionality, removal of some duplicated queue handling.

---

<details>
<summary>Checklist:</summary>

**Completeness and readability**

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with updates and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/doc/sources/contribute.rst) for details)_.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least a summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance and/or quality metrics have changed or why changes are not expected.
- [x] I have extended the benchmarking suite and provided a corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

</details>

### CODE REVIEW

**Review by david-cortes-intel — APPROVED** [review]
LGTM. Error in linear regression is not related to these changes, and error in kmeans happens in the main branch too.

File: onedal/cluster/kmeans.py [hunk 1]
```diff
@@ -30,26 +30,23 @@
...
-        tol,
-        verbose,
-        random_state,
-        n_local_trials=None,
+        init="k-means++",
+        n_init="auto",
+        max_iter=300,
+        tol=1e-4,
+        verbose=0,
+        random_state=None,
```
Comment (ethanglaser) [inline onedal/cluster/kmeans.py:49]: This would remove `copy_x` from kwargs - is it intended?

Comment (icfaust) [reply onedal/cluster/kmeans.py:49]: yep, copy_x is to match sklearn and does nothing. its not necessary and can be removed
